src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,661 @@
+        if (isinstance(image, np.ndarray) or torch.is_tensor(image)) and image.ndim == 4:
+            num_images = image.shape[0]
+
+        if num_inference_steps is None:
+            num_inference_steps = self.default_denoising_steps
+        if processing_resolution is None:
+            processing_resolution = self.default_processing_resolution
+
+        # 1. Checking inputs
+        self.check_inputs(
+            image,
+            num_inference_steps,
+            ensemble_size,
+            processing_resolution,
+            resample_method_input,
+            resample_method_output,
+            batch_size,
+            ensembling_kwargs,
+            latents,
+            generator,
+            output_prediction_format,
+            output_visualization_kwargs,
+        )
+
+        # 2. Prepare empty text conditioning. Model invocation: self.tokenizer, self.text_encoder
+        if self.empty_text_embedding is None:
+            self.encode_empty_text()
+
+        # 3. Preprocessing input image
+        image, padding, original_resolution = self.image_processor.preprocess(
+            image, processing_resolution, resample_method_input, check_input, device, dtype
+        )  # [N,3,PPH,PPW]
+
+        # 4. Encode input image into latent space. Model invocation: self.vae.encoder
+        image_latent, pred_latent = self.prepare_latent(
+            image, latents, generator, ensemble_size, batch_size
+        )  # [N*E,4,h,w], [N*E,4,h,w]
+
+        del image
+
+        batch_empty_text_embedding = self.empty_text_embedding.to(device=device, dtype=dtype).repeat(
+            batch_size, 1, 1
+        )  # [B,1024,2]
+
+        # 5. Denoising loop. Model invocation: self.unet
+        clean_latent = []
+
+        with self.progress_bar(total=num_images * ensemble_size * num_inference_steps) as progress_bar:
+            for i in range(0, num_images * ensemble_size, batch_size):
+                batch_image_latent = image_latent[i : i + batch_size]  # [B,4,h,w]
+                batch_pred_latent = pred_latent[i : i + batch_size]  # [B,4,h,w]
+                B = batch_image_latent.shape[0]
+
+                batch_text_embedding = batch_empty_text_embedding[:B]  # [B,2,1024]
+
+                self.scheduler.set_timesteps(num_inference_steps, device=device)
+
+                for t in self.scheduler.timesteps:
+                    batch_latent = torch.cat([batch_image_latent, batch_pred_latent], dim=1)  # [B,8,h,w]
+                    noise = self.unet(batch_latent, t, encoder_hidden_states=batch_text_embedding).sample  # [B,4,h,w]
+                    batch_pred_latent = self.scheduler.step(
+                        noise, t, batch_pred_latent, generator=generator
+                    ).prev_sample  # [B,4,h,w]
+                    progress_bar.update(B)
+
+                clean_latent.append(batch_pred_latent)
+
+                del batch_image_latent, batch_pred_latent, batch_text_embedding, batch_latent, noise
```

Comment:
I don't think deleting them at end of each loop achieves anything 
and we prefer to use the same variable names as much as possible so we don't create additional memory need, for example 
```python
pred_latents = []
for batch in ...:
    ....
    pred_latents.append(batch_pred_latent)
pred_latent = torch.cat(pred_latents)

```

Reply:
> I don't think deleting them at end of each loop achieves anything

Agreed.

> we prefer to use the same variable names as much as possible for example

Yes I prefer that too, but pred_latent is a read variable inside that loop.